Tribler/Core/Modules/restapi/mychannel_endpoint.py

```diff
@@ -424,15 +421,17 @@ def render_PATCH(self, request):
             request.setResponseCode(http.NOT_FOUND)
             return json.twisted_dumps({"error": "your channel has not been created"})
 
-        torrent = my_channel.get_torrent(self.infohash)
-        if not torrent:
-            request.setResponseCode(http.NOT_FOUND)
-            return json.twisted_dumps({"error": "torrent with the specified infohash could not be found"})
+        with db_session:
+            torrent = my_channel.get_torrent(self.infohash)
+            if not torrent:
+                request.setResponseCode(http.NOT_FOUND)
+                return json.twisted_dumps({"error": "torrent with the specified infohash could not be found"})
 
-        torrent.update_properties(dict([(attribute, parameters[attribute][0]) for attribute in
-                                        ['status', 'tags', 'title'] if attribute in parameters]))
+            status = torrent.update_properties(dict([(attribute, parameters[attribute][0]) for attribute in
```

Comment:
Instead of `dict([(key, a[key]) for key in something])` you can do `{key: a[key] for key in something}`, which is faster and more beautiful. In this case:

```python
status = torrent.update_properties({
    attribute: parameters[attribute][0]
    for attribute in ['status', 'tags', 'title']
    if attribute in parameters
})
```